CMakeLists.txt

```diff
@@ -26,9 +26,16 @@ if(NOT foonathan_memory_FOUND)
       " or BUILD_MEMORY_TESTS are set")
   endif()
 
-  if(BUILD_SHARED_LIBS)
-    # Library will be statically created with PIC code
-    list(APPEND extra_cmake_args -DCMAKE_POSITION_INDEPENDENT_CODE=ON)
+  # Propagate BUILD_SHARED_LIBS
+  if(DEFINED BUILD_SHARED_LIBS)
+    list(APPEND extra_cmake_args -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS})
```

Comment:
Sorry to pop up as a reviewer but as I see from the original memory repo I don't see `BUILD_SHARED_LIBS` exist over there.
Do you think this flag is doing something to the external project?

Reply:
I didn't knew it, always great to learn something new,
👍 
Thanks!

Reply:
@EduPonz 
I tried using your commit and I now get a different issue.
The install dir is hard coded and cannot be changed by the user :(
See [here](https://github.com/eProsima/foonathan_memory_vendor/blob/3b265ec02fd692d112f03cf830a36cf57516aa9f/CMakeLists.txt#L90) and [here](https://github.com/eProsima/foonathan_memory_vendor/blob/3b265ec02fd692d112f03cf830a36cf57516aa9f/CMakeLists.txt#L90)

Why is that?
That makes me not being able to use it cause my path's are different..

Reply:
The whole vendor idea is making available a dependency into your workspace locally, that is, without installing that dependency machine-wide.
That's the reason why the path is hardcoded because the user should be unaware of it.
If the user installs the dependency machine-wide (via apt for example) the vendor (through CMake introspection) will detect it and do nothing. Note that the vendor searches for a specific package thus, if the version installed is not compatible, will install the local one.
The vendor is basically a fallback mechanism.

Reply:
@MiguelBarro Thanks for the reply but something is not working for me.
I build it using `-DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}/fastdds/fastdds_install` and I get errors during Cmake configure.

```
CMake Error at foonathan_memory-config.cmake:19 (include):

-- Found foonathan_memory 

  include could not find requested file:

    C:/Work/Git/librealsense/build/third-party/foonathan_memory/../share/foonathan_memory-x64Win64VS2019/cmake/foonathan_memory-config.cmake
Call Stack (most recent call first):
  CMakeLists.txt:4 (find_package)
```

I don't have that path..

 I am trying to integrate Fast-DDS on my CMake project for 2 weeks now and I keep getting issues..
Tried:
ExternalProject_Add during build time - cannot because you concat the version name to the libs
ExternalProject_Add during configure time - cannot find foonathan (no matter what I tried)
fetchContent - failed due to errors on building `fastrtps` (see [2501)](https://github.com/eProsima/Fast-DDS/issues/2501)

Please.. Can someone share a reference project about how to properly integrate `FastDDS` into a CMake project? (Without telling the user to first install `FastDDS` and all of it's dependencies..)

I am trying to get to a point when my CMake project will integrate Fast-DDS and all of it's dependencies automatically.

Thanks,
Nir

Reply:
I deduce that you are not using colcon. Are you using CMake directly?

Reply:
Yes..

FetchContent_Declare for foonathan
and than FetchContent_Declare / externalproject_add (during configure time) for fastdds when (-THIRDPARTY*=FORCE)

If I use the original foonathan repo my FastDDS external project can't find it.
If I use eProsima (Vendor) repo, I get a configure error as mentioned above.

I couldn't find a reference online for using `FastDDS` this way, I guess most of the users are `ROS2` users..

Reply:
Hi again @Nir-Az ,

Looking at [your code](https://github.com/Nir-Az/librealsense/blob/link_fastcdr/CMake/external_fastdds.cmake#L34-L56) I think you're missing the installation step of the vendor package. As far as I can tell, following the documentation instructions (which include installing the vendor) you should be good to go. I'd advise:
* Start with clean build directory.
* Pass the installation path of Fast CDR and foonathan memory vendor to Fast DDS with `-DCMAKE_PREFIX_PATH`.